Io output staticbody number

vacary opened the pull request.

vacary committed:
[mechanics] add BodyShapeRecord as an attempt of unique pointer in Contact*R classes

vacary committed:
[io] output static body number

vacary committed:
Merge branch 'master' into io_output_staticbody_number

vacary committed:
[io] add a new table for contact info as output in the hdf5 file taht contains (id,ds1,ds2,staticbody_id)
     correct bug on the use run_options with output_contact_forces and output_contact_info

vacary committed:
[io] add  contact info as output in the hdf5 file for other Contact*R classes
     compilation without BULLET

vacary committed:
[io] fix and rename the contact_index_set options

vacary committed:
[mechanics] add debug verbose

vacary committed:
Merge branch 'master' into io_output_staticbody_number

vacary committed:
[io] output_contact_info only for bullet backend

vacary committed:
[io] correct issue in test_native_collision.py with fancy indexing in h5py

vacary commented:
@bremond I would like to have your point of   view on this PR? There is less members in the BulletR* relations classes in that way, but I need to use a map to link the pointer on the body shape record to its smart pointer. I do not know if it is a good idea ?

bremond commented:
Hy, I think that, in SiconosBulletCollisionManager, keeping track of another copy of the smart pointer is useless, as a copy is already inserted in bodyShapeMap (so the data won't go away). When you need a smart pointer from a raw pointer you can use createSprClassName as for example createSPtrbtManifoldPoint in the same file. Here you have to remove the const qualifier as bodyShapeRecordA are non const members :
```
   rel->bodyShapeRecordA = createSPtrBodyBulletShapeRecord(*const_cast<BodyBulletShapeRecord*>(pairA));
   rel->bodyShapeRecordB = createSPtrBodyBulletShapeRecord(*const_cast<BodyBulletShapeRecord*>(pairB));
```

(I need also to add DEFINE_SPTR(BodyBulletShapeRecord) in BulletSiconosFwd.hpp). This compiles, I didn't check anything.

(The createSptr* functions are defined in SiconosPointers.hpp. They symply associate an empty deleter  with a smart pointer).

vacary committed:
[mechanics] remove _bulletShapeRecordMap following @bremond comment in PR 435:

	    In SiconosBulletCollisionManager, keeping track of another copy of the smart pointer is useless, as a copy is already inserted in bodyShapeMap (so the data won't go away). When you need a smart pointer from a raw pointer you can use createSprClassName as for example createSPtrbtManifoldPoint in the same file. Here you have to remove the const qualifier as bodyShapeRecordA are non const members :

   rel->bodyShapeRecordA = createSPtrBodyBulletShapeRecord(*const_cast<BodyBulletShapeRecord*>(pairA));
   rel->bodyShapeRecordB = createSPtrBodyBulletShapeRecord(*const_cast<BodyBulletShapeRecord*>(pairB));
(I need also to add DEFINE_SPTR(BodyBulletShapeRecord) in BulletSiconosFwd.hpp). This compiles, I didn't check anything.

(The createSptr* functions are defined in SiconosPointers.hpp. They symply associate an empty deleter with a smart pointer).
